packages/neos-ui/src/Containers/Modals/InsertionMode/index.js

```diff
@@ -0,0 +1,205 @@
+                </span>)
+            });
+        }
+    }
+
+    renderNodeLabel(contextPath) {
+        const {getNodeByContextPath, nodeTypesRegistry} = this.props;
+        const node = getNodeByContextPath(contextPath);
+        const getLabel = $get('label');
+        const getNodeType = $get('nodeType');
+        const getNodeTypeLabel = compose(getLabel, nodeTypesRegistry.get.bind(nodeTypesRegistry), getNodeType);
+
+        return (
+            <span>
+                <I18n id={getNodeTypeLabel(node)}/>
+                &nbsp;"{getLabel(node)}"
+            </span>
+        );
+    }
+
+    renderTitle() {
+        const {subjectContextPath, referenceContextPath} = this.props;
+
+        return (
+            <div>
+                <Icon icon="clipboard"/>
+                <span className={style.modalTitle}>
+                    <I18n
+                        id="Neos.Neos.Ui:Main:copy__from__to--title"
+                        params={{
+                            source: this.renderNodeLabel(subjectContextPath),
+                            target: this.renderNodeLabel(referenceContextPath)
+                        }}
+                        />
+                </span>
+            </div>
+        );
+    }
+
+    renderCancel() {
+        return (
+            <Button
+                key="cancel"
+                style="lighter"
+                hoverStyle="brand"
+                onClick={this.handleCancel}
+                >
+                <I18n id="Neos.Neos:Main:cancel" fallback="Cancel"/>
+            </Button>
+        );
+    }
+
+    renderApply() {
+        return (
+            <Button
+                key="apply"
+                style="lighter"
+                hoverStyle="brand"
+                onClick={this.handleApply}
+                className={style.applyBtn}
+                >
+                <I18n id="Neos.Neos:Main:apply" fallback="Apply"/>
+            </Button>
+        );
+    }
+
+    render() {
+        const {
+            isOpen,
+            cancel,
+            subjectContextPath,
+            referenceContextPath
+        } = this.props;
+
+        if (!isOpen || !this.options.length) {
+            return null;
+        }
+
+        return (
+            <Dialog
+                actions={[this.renderCancel(), this.renderApply()]}
+                title={this.renderTitle()}
+                onRequestClose={this.handleCancel}
+                isOpen
+                >
+                <div className={style.modalContents}>
+                    <p>
+                        <I18n
+                            id="Neos.Neos.Ui:Main:copy__from__to--description"
+                            params={{
+                                source: this.renderNodeLabel(subjectContextPath),
+                                target: this.renderNodeLabel(referenceContextPath)
+                            }}
+                            />
+                    </p>
+                    <SelectBox
```

Comment:
Would be nice to change it from SelectBox into something like a group of buttons, so it would be possible to select a correct position with one click. But possibly in a different PR.